rt_gene/src/rt_gene/gaze_estimation_models_pytorch.py

```diff
@@ -211,82 +184,60 @@ def __init__(self, num_out=2):
+        eyes_x = self.concat(eyes_x)
 
-        concat = torch.cat((left_x, right_x, headpose), dim=1)
+        eyes_headpose = torch.cat((eyes_x, headpose), dim=1)
 
-        fc_output = self.classifier(concat)
+        fc_output = self.fc(eyes_headpose)
 
         return fc_output
 
 
-class GazeEstimationModelResneXt(GazeEstimationAbstractModel):
+class GazeEstimationModelPreactResnet(GazeEstimationAbstractModel):
 
     def __init__(self, num_out=2):
-        super(GazeEstimationModelResneXt, self).__init__()
-        _left_model = models.resnext50_32x4d(pretrained=True)
-        _right_model = models.resnext50_32x4d(pretrained=True)
+        super(GazeEstimationModelPreactResnet, self).__init__()
+        self.left_features = PreactResnet()
+        self.right_features = PreactResnet()
 
-        # remove the last ConvBRelu layer
-        self.left_features = nn.Sequential(
-            _left_model.conv1,
-            _left_model.bn1,
-            _left_model.relu,
-            _left_model.maxpool,
-            _left_model.layer1,
-            _left_model.layer2,
-            _left_model.layer3,
-            _left_model.layer4,
-            _left_model.avgpool
-        )
-
-        self.right_features = nn.Sequential(
-            _right_model.conv1,
-            _right_model.bn1,
-            _right_model.relu,
-            _right_model.maxpool,
-            _right_model.layer1,
-            _right_model.layer2,
-            _right_model.layer3,
-            _right_model.layer4,
-            _right_model.avgpool
-        )
-
-        for param in self.left_features.parameters():
-            param.requires_grad = True
-        for param in self.right_features.parameters():
-            param.requires_grad = True
-
-        _num_ftrs = _left_model.fc.in_features + _right_model.fc.in_features + 2  # left, right and head_pose
-        self.classifier = GazeEstimationAbstractModel._create_fc_layers(num_out=num_out, num_features=_num_ftrs)
+        self.xl, self.xr, self.concat, self.fc = GazeEstimationAbstractModel._create_fc_layers(in_features=64, out_features=num_out)
         GazeEstimationAbstractModel._init_weights(self.modules())
 
     def forward(self, left_eye, right_eye, headpose):
         left_x = self.left_features(left_eye)
         left_x = torch.flatten(left_x, 1)
+        left_x = self.xl(left_x)
 
         right_x = self.right_features(right_eye)
         right_x = torch.flatten(right_x, 1)
+        right_x = self.xr(right_x)
+
+        eyes_x = torch.cat((left_x, right_x), dim=1)
+        eyes_x = self.concat(eyes_x)
 
-        concat = torch.cat((left_x, right_x, headpose), dim=1)
+        eyes_headpose = torch.cat((eyes_x, headpose), dim=1)
 
-        fc_output = self.classifier(concat)
+        fc_output = self.fc(eyes_headpose)
 
         return fc_output
 
 
 class GazeEstimationModelMobileNetV2(GazeEstimationAbstractModel):
 
-    def __init__(self, num_out=2):  # phi, theta
+    def __init__(self, num_out=2):
```

Comment:
Shall we get rid of the MobileNetV2? Seems like there is no case in which we would want to use it.

Reply:
Is there not? Not one to try RT-GENE on mobile phone of some kind? Although to be fair, you'd need Tensorflow for that if you're gonna use it on an Android (no idea what library it would be on iOS)...

Reply:
I think it's a stretch. RT-GENE is not really made for this scenario. If someone wants to use it, they could add this model themselves. I'd personally opt for removing it, especially considering that we could simplify the `forward()` method.

Reply:
Agreed - latest commit does so.